### ISSUE TITLE
Unable to Create Account on Self-Hosted Version

### ISSUE THREAD

**Issue by ZeroGravitas23** [julianpoy/RecipeSage-selfhost#9]
I am trying to self-host this on a local linux machine with docker and seem to be having issues with account creation.

It is saying: "An unexpected error has occurred. Please try again" on the account creation page.

When looking into the Static Recipe Sage container I see this error message: 

`2021/04/06 16:05:26 [error] 22#22: *2 open() "/usr/share/nginx/html/api/versioncheck" failed (2: No such file or directory), client: 172.24.0.1, server: localhost, request: "GET /api/versioncheck?version=v2.7.3-beta4 HTTP/1.1", host: "sage.****.duckdns.org", referrer: "https://sage.****.duckdns.org/"

2021/04/06 16:06:03 [error] 22#22: *3 open() "/usr/share/nginx/html/api/users/register" failed (2: No such file or directory), client: 172.24.0.1, server: localhost, request: "POST /api/users/register HTTP/1.1", host: "sage.****.duckdns.org", referrer: "https://sage.****.duckdns.org/"`

Also to specify - in the compose file I removed the nginx section because I use Nginx Proxy Manager which handles the proxies for all webapps running on the machine. Not sure if that is what might be causing the issue but it seems like it's looking for some nginx filepath.

Any help is greatly appreciated in understanding what's going on. Thank you for the great work on this app by the way, very well designed and great to see there is a self-host option!

**Comment by DiversityBug** [comment]
Hello Julian, 
You did an **outstanding job** with the online version already. Now having the peace of mind that there is a self-host option too is great.
I am observing the same behavior. Though I did not change the compose, I see the recipesage_proxy_1 failed to create.
Please find the log and screenshot attached.
Let me know if you ever think of going multi lingual with your app to help.
**Thanks for your great work!** 

![docker compose error](https://user-images.githubusercontent.com/82063721/113787802-385dee80-972b-11eb-8f7c-66d6a7a4a49e.png)
[recipesage_static_1.txt](https://github.com/julianpoy/RecipeSage-selfhost/files/6267924/recipesage_static_1.txt)

**Comment by Typhonragewind** [comment]
Can you try to manually create the proxy.conf file at the host designed location before starting up the compose? I've had a similar, if not equal error in the past and that resolved it.

**Comment by julianpoy** [comment]
The first two comments in this chain aren't related - going to only address the first comment here for now.

> Also to specify - in the compose file I removed the nginx section because I use Nginx Proxy Manager which handles the proxies for all webapps running on the machine. Not sure if that is what might be causing the issue but it seems like it's looking for some nginx filepath.

This is why it doesn't work. The nginx proxy is necessary for routing the various application services and shouldn't be removed.

**Comment by ZeroGravitas23** [comment]
Thanks for the response Julian. That makes sense and at first I wanted to see if I could use Nginx Proxy Manager to replace the proxy in the web app here, obviously that wasn't working. 

I did change my config and set up the nginx proxy as described. Funny enough though I am still getting the inability to create an account (same "An unexpected error occurred. Please try again." error)

Here is my docker compose for this app:
https://github.com/ZeroGravitas23/recipesage/blob/5fbb2323532daaba6ea94b6fd1e460b26c0f796c/recipe_sage%20compose%20(210412).yml#L1-L174

Here is the nginx proxy I have for the app:
https://github.com/ZeroGravitas23/recipesage/blob/0d4d9ebb80311851345caf344abfbf46490045d0/default.conf#L1-L34

And checking the proxy here is the error I am seeing:
`2021/04/12` 15:21:43 [error] 30#30: *3 connect() failed (111: Connection refused) while connecting to upstream, client: 10.10.10.10, server: localhost, request: "POST /api/users/register HTTP/1.1", upstream: "http://172.30.0.10:3100/users/register", host: "10.10.11.1:83", referrer: "http://10.10.11.1:83/"'

Unsure of why it can't reach the /api/ piece as they are all on the same networks except for the db which is only on the backend. Let me know your thoughts on what might be going on.

**Comment by julianpoy** [comment]
Both of those links 404 for me - are you sure they're public?

**Comment by ZeroGravitas23** [comment]
My mistake, was set to private initially. They are accessible now.

**Comment by julianpoy** [comment]
This is your issue: 
https://github.com/ZeroGravitas23/recipesage/blob/0d4d9ebb80311851345caf344abfbf46490045d0/default.conf#L27

That port number should be 3000. Keep in mind that container-to-container networking doesn't care about port forwarding ([such as you've defined here](https://github.com/ZeroGravitas23/recipesage/blob/5fbb2323532daaba6ea94b6fd1e460b26c0f796c/recipe_sage%20compose%20(210412).yml#L74))

**Comment by ZeroGravitas23** [comment]
Thank you Julian for getting back to me. And that was the exact issue. It was the set of ports I tried to use that were not mapping through the containers correctly. I have reset all ports back to their defaults and it works wonderfully now.

Also, it was very easy then to map to Caddy which serves this out externally without any issue (on port 80 for the Sage Ngnix).

Right now the only thing I have changed compared to your compose file are the names of the containers which I have updated everywhere they are used/mentioned accordingly including Nginx. This is mainly to keep this App and it's containers separate from some other webapps that are running on the same machine. I don't think this will cause too much grief if updates come through but I will test as needed.

Thanks for the help and explanation on this issue.
